File: TeachersGuardAPI/App/UseCases/Place/PlaceUseCase.cs
using TeachersGuardAPI.App.DTOs.Place;
using TeachersGuardAPI.Domain.Repositories;
using TeachersGuardAPI.Infraestructure.Mappers;

namespace TeachersGuardAPI.App.UseCases.PlaceUseCase
{
    public class PlaceUseCase
    {
        private readonly IPlaceRepository _placeRepository;
        public PlaceUseCase(IPlaceRepository placeRepository) 
        {
            _placeRepository = placeRepository;
        }

        public async Task<PlaceDto?> GetPlaceByPlaceId(string placeId)
        {
            var place = await _placeRepository.GetPlaceByPlaceId(placeId);

            if (place == null) return null;

            return PlaceMapper.MapPlaceEntityToPlaceDto(place);
        }

    }
}

File: lib/useCases/place/place_use_case.dart
import 'package:teachersguard/domain/repositories/repositories.dart';

import '../../domain/entities/entities.dart';

class PlaceUseCase {
  final PlaceRepository _placeRepository;

  PlaceUseCase(this._placeRepository);

  Future<List<Place>> getAllPlaces() {
    return _placeRepository.fetchPlaces();
  }

  Future<WeekUsePlace> getUsePlaceByPlaceId(String placeId) {
    return _placeRepository.fetchUsePlace(placeId);
  }
}
